### DISCUSSION TITLE
How to read content in Firestore 'map' ?

### DISCUSSION THREAD

**Discussion by Baylung** [SchneiderInfosystems/FB4D#74] [Q&A] [answered]
I'm using Google add-on ("MailOut") for sending emails.
After sending email it creates documents with the following content:

![image](https://user-images.githubusercontent.com/18363588/112610086-e1186e00-8e4e-11eb-86b0-a833554a66f6.png)

`delivery` is the 'map' field. I need to read `status` and `endTime` from this `delivery` map.
`template` is another 'map' field. I need to read `name` from this `template` map.

How I can do it? I tried this and that, but getting something that can't use... or getting exectins. 

Please advise how to find all fields by name in a 'map' ? 

For example, I have `Docs.Document(0)` as result of query. How I can get to the 
`Docs.Document(0)->delivery map->status field value`
`Docs.Document(0)->delivery map->endTime field value`
`Docs.Document(0)->template map->name field value`
?

Thank you

**Comment by SchneiderInfosystems** [comment, marked as answer]
Hi 

Sorry for the late answer.

Do you have checked the method `TfmxFirebaseDemo.ShowDocument(Doc: IFirestoreDocument). It demonstrates how to access all kind of fields within a document.

Christoph

**Reply by Baylung** [reply]
Thanks for your reply! I went through your Demo sources and learned from there how to read Map fields.
And thank you for the amazing FB4D. Hope you will expand it more in the future :) 

(Just a comment after looking at Map:  comparing to a very convenient way of accessing non-map fields (GetFieldValue, Get***ValueDev) the way of how to access Map fields maybe could be made more convenient to...)
